scipy/signal/tests/test_signaltools.py

```diff
@@ -748,62 +763,872 @@ def test_many_sizes(self, n):
         out = fftconvolve(a, b, 'full', axes=[0])
         assert_allclose(out, expected, atol=1e-10)
 
-    def test_invalid_shapes(self):
+
+def fftconvolve_err(*args, **kwargs):
+    raise RuntimeError('Fell back to fftconvolve')
+
+
+class TestOAConvolve(object):
+    @pytest.mark.slow()
+    @pytest.mark.parametrize('shape_a_0', list(range(100)) +
+                             list(range(100, 1000, 23)))
+    @pytest.mark.parametrize('shape_b_0', list(range(100)) +
+                             list(range(100, 1000, 23)))
+    def test_real_manylens(self, shape_a_0, shape_b_0):
+        if shape_a_0 <= shape_b_0:
+            return
+
+        a = np.random.rand(shape_a_0)
+        b = np.random.rand(shape_b_0)
+
+        expected = fftconvolve(a, b)
+        out = oaconvolve(a, b)
+
+        assert_array_almost_equal(out, expected)
+
+    @pytest.mark.parametrize('shape_a_0', [50, 49, 6, 4])
+    @pytest.mark.parametrize('shape_b_0', [50, 49, 6, 4])
+    def test_real_noaxes(self, shape_a_0, shape_b_0, monkeypatch):
+        if shape_a_0 == shape_b_0:
+            return
+        if abs(shape_a_0-shape_b_0) <= 2:
+            return
+
+        a = np.random.rand(shape_a_0)
+        b = np.random.rand(shape_b_0)
+
+        expected = fftconvolve(a, b)
+
+        monkeypatch.setattr(signal.signaltools, 'fftconvolve',
+                            fftconvolve_err)
+        out = oaconvolve(a, b)
+
+        assert_array_almost_equal(out, expected)
+
+    @pytest.mark.parametrize('axes', [None, 0, [0], -1, [-1]])
+    @pytest.mark.parametrize('shape_a_0', [50, 49, 6, 4])
+    @pytest.mark.parametrize('shape_b_0', [50, 49, 6, 4])
+    def test_real(self, axes, shape_a_0, shape_b_0, monkeypatch):
+        if shape_a_0 == shape_b_0:
+            return
+        if abs(shape_a_0-shape_b_0) <= 2:
+            return
+
+        a = np.random.rand(shape_a_0)
+        b = np.random.rand(shape_b_0)
+
+        expected = fftconvolve(a, b, axes=axes)
+
+        monkeypatch.setattr(signal.signaltools, 'fftconvolve',
+                            fftconvolve_err)
+        out = oaconvolve(a, b, axes=axes)
+
+        assert_array_almost_equal(out, expected)
+
+    @pytest.mark.parametrize('axes', [0, [0], -2, [-2]])
+    @pytest.mark.parametrize('shape_a_0', [100, 99, 6, 4])
```

Comment:
`99` was selected for that reason, and I test a huge number of values in `test_real_manylens`.  Do you have other values you think I should test?  I don't want to add too many since it causes the testing time to grow considerably.

Reply:
`99 = 11 * 3 * 3` is now a fast length for the complex FFT. 47 should be good now though.